src/main/java/com/adamk33n3r/runelite/watchdog/EventHandler.java

```diff
@@ -189,26 +191,54 @@ private void onItemContainerChanged(ItemContainerChanged itemContainerChanged) {
-                if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.FULL && itemCount == 28) {
-                    this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
-                } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.EMPTY && itemCount == 0) {
-                    this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
-                } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
-                    Map<Integer, Integer> allItems = new HashMap<>();
-                    Arrays.stream(items)
-                        .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
-                    allItems.entrySet().stream()
-                        .filter(itemWithCount -> inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity())
-                        .map(itemWithCount -> this.matchPattern(inventoryAlert, this.itemManager.getItemComposition(itemWithCount.getKey()).getName()))
-                        .filter(Objects::nonNull)
-                        .findFirst()
-                        .ifPresent(groups -> this.fireAlert(inventoryAlert, groups));
-                }
-            });
+        Item[] items = itemContainerChanged.getItemContainer().getItems();
+        long itemCount = Arrays.stream(items).filter(item -> item.getId() > -1).count();
+        Map<Integer, Integer> allItems = new HashMap<>();
+        Arrays.stream(items)
+                .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
+        // Skip firing alerts if there are no previous items, since we just logged in. Even an empty inventory will have a map of -1 itemIds.
+        if (!this.previousItemsTable.isEmpty()) {
+            this.alertManager.getAllEnabledAlertsOfType(InventoryAlert.class)
+                    .forEach(inventoryAlert -> {
+                        if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.FULL && itemCount == 28) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.EMPTY && itemCount == 0) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
+                            allItems.entrySet().stream()
+                                    .filter(itemWithCount -> itemWithCount.getKey() != -1 && (inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity()))
```

Comment:
If we filter out empty items, then we'd need another variable to track and skip the first inventory update of the session. I'm using the fact that there's a difference in `this.previousItemsTable` between an empty inventory (which will be a Map with 1 key) and a null inventory (which will be an empty Map) to skip the first update to avoid alerts firing upon login.

Reply:
Aren't you ONLY checking if `previousItemsTable` is empty? Wouldn't that check still work even if `allItems` is empty too?

Reply:
Like upon login, `previousItemsTable` will be empty. If we add that filter here, then `allItems` will also be empty. Then you check `if (!this.previousItemsTable.isEmpty())` which would return false as desired and we'd skip the first instance.

Reply:
Oh I see now, ya. Hm....ok. Good enough reason for me.